flask/app.py

```diff
@@ -1370,6 +1390,33 @@ def url_defaults(self, f):
         self.url_default_functions.setdefault(None, []).append(f)
         return f
 
+    def _find_error_handler(self, e):
+        """Finds a registered error handler for the request’s blueprint.
+        If neither blueprint nor App has a suitable handler registered, returns None
+        """
+        exc_class, code = self._get_exc_class_and_code(type(e))
+        
+        def find_superclass(d):
+            if not d:
+                return None
+            for superclass in exc_class.mro():
+                if superclass is BaseException:
+                    return None
+                handler = d.get(superclass)
+                if handler is not None:
+                    d[exc_class] = handler  # cache for next time exc_class is raised
```

Comment:
well, it’s just two lines and doesn’t do any harm, since the registered handlers may not be changed after app start (so the cache will never be invalidated)

and the few bytes more memory are even less significant than the cycles saved ;)